compiler/pipes/optimization.cpp

```diff
@@ -218,6 +218,50 @@ VertexPtr OptimizationPass::remove_extra_conversions(VertexPtr root) {
   return root;
 }
 
+// The function will try to convert the expression to a call to the
+// __toString method if the expression has a class type and the class
+// has such a method.
+// If the conversion fails, an empty VertexPtr will be returned.
+VertexPtr OptimizationPass::try_convert_expr_to_call_to_string_method(VertexPtr expr) {
+  const auto *type = tinf::get_type(expr);
+  if (type == nullptr) {
+    return {};
+  }
+
+  const auto klass = type->class_type();
+  if (!klass) {
+    return {};
+  }
+
+  if (!klass->has_tostring) {
```

Comment:
It seems to me, that we don't need this flag, because we can use a result of `get_instance_method`:
```cpp
const auto *to_string_method = klass->get_instance_method(ClassData::NAME_OF_TO_STRING);
if (!to_string_method) {
  return error;
}
  ```

Reply:
Removed.